tests/backup/backup_azure_credential_test.go

```diff
@@ -0,0 +1,231 @@
+		backupLocationMandatoryFieldsUID                       string
+		tenantID                                               string
+		clientID                                               string
+		clientSecret                                           string
+		subscriptionID                                         string
+		accountName                                            string
+		accountKey                                             string
+		appNamespaces                                          []string
+		controlChannel                                         chan string
+		backupLocationMap1                                     map[string]string
+		cloudCredentialMap1                                    map[string]string
+		backupLocationMap2                                     map[string]string
+		cloudCredentialMap2                                    map[string]string
+		errorGroup                                             *errgroup.Group
+		azureConfigFields                                      *api.AzureConfig
+		scheduledAppContexts                                   []*scheduler.Context
+		contexts                                               []*scheduler.Context
+		appContexts                                            []*scheduler.Context
+	)
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("AzureCloudAccountCreationWithMandatoryAndNonMandatoryFields", "Azure cloud account with mandatory and non mandatory fields", nil, 31661, Sagrawal, Q2FY25)
+		backupLocationMap1 = make(map[string]string)
+		cloudCredentialMap1 = make(map[string]string)
+		backupLocationMap2 = make(map[string]string)
+		cloudCredentialMap2 = make(map[string]string)
+		log.InfoD("Deploying applications required for the testcase")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", TaskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = AppReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+			}
+		}
+	})
+
+	It("Azure cloud account with mandatory and non mandatory fields", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		tenantID, clientID, clientSecret, subscriptionID, accountName, accountKey = GetAzureCredsFromEnv()
+		Step("Validating applications", func() {
+			log.InfoD("Validating applications")
+			ctx, _ := backup.GetAdminCtxFromSecret()
+			controlChannel, errorGroup = ValidateApplicationsStartData(scheduledAppContexts, ctx)
+		})
+
+		Step("Creating azure cloud account and backup location with all fields", func() {
+			backupLocationProviders := GetBackupProviders()
+			for _, provider := range backupLocationProviders {
```

Comment:
How about initializing the backupLocationProvider to Azure in the JustBeforeEach ?

Reply:
Done.Since its valid only for azure ,removed the loop based on backupLocationProvider